src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php

```diff
@@ -207,14 +211,42 @@ private function doQueryForNonValue( PageRequestOptions $pageRequestOptions, Req
 			$requestOptions
 		);
 	}
+	
+	private function destructureDIContainer( $DIProperty, $dataItem, $pageRequestOptions ) {
+		$multiValue = DataValueFactory::getInstance()->newDataValueByItem(
+			$dataItem,
+			$DIProperty
+		);
+		$properties = $multiValue->getPropertyDataItems( $property );
+
+		// Property for the reference value
+		$DIProperty = $properties[0];
```

Comment:
Reusing a variable name like this isn't great

Reply:
I've renamed the variable and read the blog page. In this case I wasn't able to find a better way. I've added a comment with some explanation. An alternate way would be to create the correct dataitem manually using `DataValueFactory` however the class `PageRequestOptions` does some additional operation like unescaping the data value. I'm not sure if that can be avoided in this particular context.